custom_components/wyzeapi/number.py

```diff
@@ -0,0 +1,176 @@
+    def name(self) -> str:
+        """Return the name of the zone quickrun duration."""
+        return f" {self._zone.name}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID for the zone quickrun duration."""
+        return f"{self._device.mac}-zone-{self._zone.zone_number}-quickrun-duration"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device.mac)},
+            name=self._device.nickname,
+            manufacturer="WyzeLabs",
+            model=self._device.product_model,
+            serial_number=self._device.sn,
+            connections={(dr.CONNECTION_NETWORK_MAC, self._device.mac)},
+        )
+
+    @property
+    def native_value(self) -> float:
+        """Return the current value."""
+        return float(self._zone.quickrun_duration)
+
+    @property
+    def native_min_value(self) -> float:
+        """Return the minimum value."""
+        return 1.0
+
+    @property
+    def native_max_value(self) -> float:
+        """Return the maximum value."""
+        return 3600.0  # 1 hour
+
+    @property
+    def native_step(self) -> float:
+        """Return the step value."""
+        return 1.0
+
+    @property
+    def mode(self) -> str:
+        """Return the mode of the number entity."""
+        return "box"
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return the unit of measurement."""
+        return "s"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon for the quickrun duration number."""
+        return "mdi:timer"
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the value."""
+        await self._irrigation_service.set_zone_quickrun_duration(
+            self._device,
+            self._zone.zone_number,
+            int(value)
+        )
+        self._zone.quickrun_duration = int(value)
+        self.async_write_ha_state()
+
+    async def _async_load_value(self) -> None:
+        """Load the value from Home Assistant state or update from irrigation service."""
+        # Try to get the last number data from Home Assistant
+        state = await self.async_get_last_number_data()
+        if state and state.native_value is not None:
+            try:
+                self._zone.quickrun_duration = int(state.native_value)
+                return
+            except (ValueError, TypeError):
+                pass
+        
+        # If no valid state exists, update from irrigation service
+        self._device = await self._irrigation_service.update(self._device)
+        for zone in self._device.zones:
+            if zone.zone_number == self._zone.zone_number:
+                self._zone = zone
+                break
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to updates."""
+        await self.async_get_last_number_data()
```

Comment:
Isn't this redundant with L127? Seems like you can just call `_async_load_value `and that will get the last number data.

Reply:
ill check this and `async_will_remove_from_hass`. i thought in my testing they were required, but that might have been before i moved to using a `RestoreNumber`

Reply:
yeah this was a dupe from above